kci_test

```diff
@@ -73,42 +77,114 @@ class cmd_validate(Command):
         return True
 
 
+def _iterate_fs_configs(file_systems_config):
+    for fs, config in file_systems_config.items():
+        fs_parts = fs.split('_')
+        if fs_parts[0] == 'debian':
+            name = fs_parts[1]
+            root_type = fs_parts[2]
+        elif fs_parts[0].startswith('buildroot'):
+            name = fs_parts[0]
+            root_type = fs_parts[1]
+        else:
+            continue
+        yield fs, config, name, root_type
+
+
+def _check_rootfs_urls(fs_config, rootfs_config, root_type, verbose):
+    url_formats = list(
+        url_fmt for url_fmt in (
+            fs_config.get_url_format(fmt_type)
+            for fmt_type in set({root_type}).union({'nfs', 'ramdisk'})
+        ) if url_fmt
+    )
```

Comment:
I think we can just use a normal list comprehension here

```suggestion
    url_formats = [
        url_fmt for url_fmt in (
            fs_config.get_url_format(fmt_type)
            for fmt_type in set({root_type}).union({'nfs', 'ramdisk'})
        ) if url_fmt
    ]
```

The internal one can be also converted to list comprehension or you can leave it as a generator expression as it is now. Having said that there's not much gain from using genexp there.

Reply:
Yes I think that's some leftover from previous experiments, I'll change it to `[]`